### ISSUE TITLE
RadioButton: Cannot be selected with screenreader on Safari

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#1571]
**Component**
Radio Button

**Description of bug**
Unexpected behavior when using a screen reader and Safari: cannot select options

**Steps to reproduce**
On https://react-magma.cengage.com/version/4.6.0/api/radio/ and https://next--upbeat-sinoussi-f675aa.netlify.app/api/radio/, and on every other example:

1. Open basic usage;
2. Turn on screen reader;
3. Focus on the first radio button;
4. Try to move focus using arrows;
5. Unexpected behavior;

**Note**:
It works fine if we hold `Shift`;

**Can it only be reproduced in dev or is it reproducible in the last released version?**
Both

**Reproducible on**
- Sequoia 15 - Safari 18 - VoiceOver
- Ubuntu 22.04.4 LTS - Chrome Version 124.0.6367.118 - Default Ubuntu's screen reader

**Notes**
If I navigate to https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_radio for plain old HTML radio buttons on Safari + screenreader, I still don't know how to select items and properly navigate through them

**Comment by olehnoskov-cengage** [comment]
During the investigation of this issue, other issues were found in the following components:
**Checkbox**
**Intermediate** Checkbox
**Toogle**
